craft_application/services/project.py

```diff
@@ -297,8 +297,8 @@ def partitions(self) -> list[str] | None:
             )
         return self.__partitions
 
-    @staticmethod
     def _app_preprocess_project(
+        self,
```

Comment:
I'm nervous about doing this since it makes it too easy to fall into the trap of reading class state in this method when it should be stateless.

I've proposed an alternative to you on Matrix. If that's not feasible, please add some clarifications to the docstring here about how this is expected to be used (don't use class state, keep side effects to a minimum, etc.). Most directly I can see people accidentally using `self._build_on` etc. rather than the values passed to this method.

Reply:
Agreed.  I'll move this to a draft and test the solution proposed in Matrix:
* Pop `parse-info` in `_app_preprocess_project`.
* Add a `get_parse_info` to the project service that loads the raw project.